Scripts/Outland: Added Shartuul's Transporter event

14shagov opened the pull request:
<!--- (**********************************)
      (** Fill in the following fields **)
      (**********************************) --->

**Changes proposed:**

-  Added Shartuul's Transporter event

**Issues addressed:**

Closes https://github.com/TrinityCore/TrinityCore/issues/18655


**Tests performed:**

Builded and tested in game


**Known issues and TODO list:**

- ~~I can't find the portal from which Shartuul exits.~~
- ~~Spell **40675** is not working.~~
- ~~Before Eye Stalk spawns, Shartuul summons a big meteor that hits the crystal, can't find this spell.~~
- Spell **41550** doesn't work right (directs the beam under itself, when it should be on the enemy)
- ~~Before Eye Stalk appears, small meteorites are summoned. I can't implement it.~~
- ~~Eye of Shartuul, when using Disruption Ray, causes Stun Fields, which are affected by the corresponding animation, I can’t find this animation.~~
- Spell **40493** is not working.
- ~~When replacing a demon, the action bar of skills does not change; you need to click on the green arrow, which changes the bars.~~
- ~~Sometimes, for some reason, when trash mobs appear, two green lightning bolts are thrown, there should be one~~
- I don’t know how to implement the change of abilities for Shivan Assassin when changing aspect
- ~~When the bosses Doom Punisher, Shivan Assassin and Shartuul appear, they are in some kind of shadow. I can't find a spell that can do this.~~
- Not sure about the flag_extra changes. But if you don't remove flag 128, the creatures won't aggro. Yes, DB has similar creatures that are already transformed and don't have unnecessary flag_extra, but they are completely different in characteristics.
- ~~Madness Rift should only cast in newly created Eye stalk.~~
- ~~Charm - North 2-3 (40382,40523) - when used, cast Character Invis and Immune (40357), which makes all creatures in the area invisible, and Possess has no effect on Demon.~~
<!--- Notes
- Enable the setting "[√] Allow edits from maintainers." when creating your pull request.
- If this PR only contains SQL files, open a new issue instead and post or link the SQL in the issue.
- When adding new SQL files, name them 9999_99_99_99_db_name.sql to reduce the chance of possible merge conflicts.
--->

Commit:
Shartuul's Transporter event

14shagov commented:
Issues:
I can't find the portal from which Shartuul exits.
Spell 40675 is not working.
Before Eye Stalk spawns, Shartuul summons a meteor that hits the crystal, can't find this spell.
Spell 41550 doesn't work right (directs the beam under itself, when it should be on the enemy)
Before Eye Stalk appears, small meteorites are summoned. I can't find this spell
Eye of Shartuul, when using Disruption Ray, causes Stun Fields, which are affected by the corresponding animation, I can’t find this animation.
Spell 40493 is not working.
When replacing a demon, the action bar of skills does not change; you need to click on the green arrow, which changes the bars.
Sometimes, for some reason, when trash mobs appear, two green lightning bolts are thrown, there should be one

Faq commented:
add that under: `Known issues and TODO list:`

Commit:
fix

lineagedr commented:
Pog

heyitsbench commented:
I do have sniffs of a few attempts at this event if it'd be any help to you. I can send them through Discord if you have it: @heyitsbench.

14shagov committed:
Fixed spells 40675 and 41550

Co-authored-by: 14shagov <14shagov>

14shagov commented:
This commit fixes issues with spells 40675 and 40493. I misspelled it.

14shagov committed:
Move possession_transfer logic to spell script, add stationary_defenderAI, SPELL_MADNESS_RIFT and SPELL_TOUCH_OF_MADNESS (#4)

14shagov commented:
- It turns out that ability **40493** (Super Jump) still doesn't work. If you use it under Doom Punisher control, the ability can't select a target.
- A new issue occurred when I replaced Charm (**530**) with Charm - North 2-3 (**40382**, **40523**) - when used, cast Character Invis and Immune (**40357**), which makes all creatures in the area invisible, and Possess has no effect on Demon.
- Another new issue, Madness Rift must only cast in newly created Eye stalk.

14shagov committed:
Merge branch '3.3.5' into Shartuul's-Transporter-event

14shagov committed:
fix

Reworked the logic associated with Shield Lightning and Spawn Lightning

14shagov committed:
fix

- Madness Rift now only casts in newly summoned Eye stalk.
- I completely canceled the first effect of Charm - North 2-3 (40382,40523), which fixed the demon possession bug. But this problem should probably be handled in a different way.

14shagov commented:
I'm wondering if I should replace all npc and boss classes with structs? And use `RegisterCreatureAI` instead of `new npc_blabla`. As far as I understand, the new standard does exactly that.

14shagov committed:
fix

Thrash Waves Optimization

14shagov committed:
Update to current standards (#8)

Update to current standards

14shagov committed:
Merge branch '3.3.5' into Shartuul's-Transporter-event

14shagov committed:
fix

Reworking the mechanics of event failure

Aokromes committed:
Merge branch '3.3.5' into Shartuul's-Transporter-event
